middleware/request-validation.ts

```diff
@@ -1,15 +1,15 @@
 import { type NextFunction, type Request, type Response } from 'express'
-import { validationResult } from 'express-validator'
+import { checkSchema, validationResult, type Schema } from 'express-validator'
+import { type DefaultSchemaKeys } from 'express-validator/src/middlewares/schema'
 
-export const validateRequest = (validation: any[]): any => {
-  const flattenedValidationArrays = validation.flat(Infinity)
-  const validationArray = [...flattenedValidationArrays]
-  validationArray.push((req: Request, res: Response, next: NextFunction) => {
-    const errors = validationResult(req)
-    if (!errors.isEmpty()) {
-      return res.status(400).json({ errors: errors.array() })
-    }
-    next()
-  })
-  return validationArray
+export const validateRequest = (validation: Schema<DefaultSchemaKeys>) => async (req: Request, res: Response, next: NextFunction) => {
+  const validationChain = checkSchema(validation)
+  for (const validation of validationChain) {
```

Comment:
tutaj taki niuans ale w przypadku operacji ktorych wynik nie wplywa na nic co jest za nimi czasem lepiej jest uzyc Promise.all ze wzgledu na to ze wykonuja sie rownolegle i co do zasady dzialaja szybciej niz for const of

Reply:
https://stackoverflow.com/questions/45285129/any-difference-between-await-promise-all-and-multiple-await